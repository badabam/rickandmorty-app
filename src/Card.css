.Card {
  padding: 20px;
  background: #eee;
  box-shadow: 0 4px 8px #000a;
}
