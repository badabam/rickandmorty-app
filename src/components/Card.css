.Card {
  display: grid;
  gap: 10px;
  padding: 20px;
  background: #dedede;
  box-shadow: 0 4px 8px #0006;
  border-radius: 4px;
}

.Card img {
  width: 100%;
  height: auto;
}
