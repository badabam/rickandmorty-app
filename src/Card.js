import './Card.css'

export default function Card({ name, species, image }) {
  return (
    <section className="Card">
      {name} ({species})
      <img src={image} alt="" />
    </section>
  )
}
