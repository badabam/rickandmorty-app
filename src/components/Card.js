import { useState } from 'react'
import './Card.css'

export default function Card({
  name,
  species,
  image,
  status,
  origin,
  location,
  gender,
}) {
  const icon = species === 'Human' ? 'ðŸ‘¤' : 'ðŸ‘½'
  const [areDetailsVisible, setAreDetailsVisible] = useState(false)

  return (
    <section className="Card">
      <h2>
        {icon} {name}
      </h2>
      <img data-testid="card-image" src={image} alt="" />
      <button onClick={() => setAreDetailsVisible(!areDetailsVisible)}>
        {areDetailsVisible ? 'Hide details' : 'Show details'}
      </button>
      <dl data-testid="card-details" hidden={!areDetailsVisible}>
        <dt>Gender</dt>
        <dd>{gender}</dd>
        <dt>Status</dt>
        <dd>{status}</dd>
        <dt>Origin</dt>
        <dd>{origin}</dd>
        <dt>Location</dt>
        <dd>{location}</dd>
      </dl>
    </section>
  )
}
