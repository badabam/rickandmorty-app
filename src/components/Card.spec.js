import { fireEvent, render, screen } from '@testing-library/react'
import Card from './Card'

describe('Card', () => {
  it('should render a card component from props', () => {
    const { container } = render(
      <Card
        name="Jane"
        species="Human"
        image="https://source.unsplash.com/300x300/?goat"
        status="alive"
        origin="Earth"
        location="Moon"
        gender="Women"
      />
    )

    expect(screen.getByText('Jane', { exact: false })).toBeInTheDocument()
    expect(screen.getByText('ğŸ‘¤', { exact: false })).toBeInTheDocument()
    expect(screen.getByTestId('card-image')).toBeInTheDocument()
    expect(screen.getByText('alive')).toBeInTheDocument()
    expect(screen.getByText('Earth')).toBeInTheDocument()
    expect(screen.getByText('Moon')).toBeInTheDocument()
    expect(screen.getByText('Women')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show the card if the button was clicked', () => {
    const { container } = render(
      <Card
        name="Jane"
        species="Human"
        image="https://source.unsplash.com/300x300/?goat"
        status="alive"
        origin="Earth"
        location="Moon"
        gender="Women"
      />
    )

    const hiddenElement = screen.getByTestId('card-details')
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(hiddenElement).not.toHaveAttribute('hidden')
  })

  it('should render "ğŸ‘¤" if the species is "Human"', () => {
    const { container } = render(
      <Card
        name="Jane"
        species="Human"
        image="https://source.unsplash.com/300x300/?goat"
        status="alive"
        origin="Earth"
        location="Moon"
        gender="Women"
      />
    )

    expect(screen.getByText('ğŸ‘¤', { exact: false })).toBeInTheDocument()
  })

  it('should render "ğŸ‘½" if the species it not "Human"', () => {
    const { container } = render(
      <Card
        name="Jane"
        species="Alien"
        image="https://source.unsplash.com/300x300/?goat"
        status="alive"
        origin="Earth"
        location="Moon"
        gender="Women"
      />
    )

    expect(screen.getByText('ğŸ‘½', { exact: false })).toBeInTheDocument()
  })
})
