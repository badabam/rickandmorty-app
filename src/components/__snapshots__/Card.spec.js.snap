// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card component from props 1`] = `
<section
  class="Card"
>
  <h2>
    ðŸ‘¤
     
    Jane
  </h2>
  <img
    alt=""
    data-testid="card-image"
    src="https://source.unsplash.com/300x300/?goat"
  />
  <button>
    Show details
  </button>
  <dl
    data-testid="card-details"
    hidden=""
  >
    <dt>
      Gender
    </dt>
    <dd>
      Women
    </dd>
    <dt>
      Status
    </dt>
    <dd>
      alive
    </dd>
    <dt>
      Origin
    </dt>
    <dd>
      Earth
    </dd>
    <dt>
      Location
    </dt>
    <dd>
      Moon
    </dd>
  </dl>
</section>
`;
